tests/pipelines/blipdiffusion/test_blipdiffusion.py

```diff
@@ -0,0 +1,195 @@
+        }
+        blip_config = Blip2Config(vision_config=vision_config, qformer_config=qformer_config, num_query_tokens=16)
+        qformer = Blip2QFormerModel(blip_config)
+        return qformer
+
+
+    @property
+    def dummy_unet(self):
+        torch.manual_seed(0)
+        unet = UNet2DConditionModel(
+            block_out_channels=(32, 64),
+            layers_per_block=2,
+            sample_size=32,
+            in_channels=4,
+            out_channels=4,
+            down_block_types=("DownBlock2D", "CrossAttnDownBlock2D"),
+            up_block_types=("CrossAttnUpBlock2D", "UpBlock2D"),
+            cross_attention_dim=32,
+        )
+        return unet
+
+    def get_dummy_components(self):
+        text_encoder= self.dummy_text_encoder  
+        vae = self.dummy_vae
+        qformer= self.dummy_qformer
+        unet = self.dummy_unet
+        tokenizer = CLIPTokenizer.from_pretrained("ayushtues/blipdiffusion", subfolder="tokenizer", cache_dir='./cache')
+        scheduler = PNDMScheduler(
+            beta_start=0.00085,
+            beta_end=0.012,
+            beta_schedule="scaled_linear",
+            set_alpha_to_one=False,
+            skip_prk_steps=True,
+        )
+
+        components = {
+            "text_encoder": text_encoder,
+            "vae": vae,
+            "qformer": qformer,
+            "unet": unet,
+            "tokenizer": tokenizer,
+            "scheduler": scheduler,
+        }
+        return components
+
+    def get_dummy_inputs(self, device, seed=0):
+        reference_image = np.random.rand(224,224, 3) * 255
+        reference_image = Image.fromarray(reference_image.astype('uint8')).convert('RGBA')
+
+        if str(device).startswith("mps"):
+            generator = torch.manual_seed(seed)
+        else:
+            generator = torch.Generator(device=device).manual_seed(seed)
+        inputs = {
+            "prompt": "swimming underwater",
+            "reference_image": reference_image,
+            "source_subject_category": "dog",
+            "target_subject_category": "dog",
+            "height": 512,
+            "width": 512,
+            "guidance_scale": 7.5,
+            "num_inference_steps": 2,
+        }
+        return inputs
+
+    def test_blipdiffusion(self):
+        device = "cpu"
+        components = self.get_dummy_components()
+
+        pipe = self.pipeline_class(**components)
+        pipe = pipe.to(device)
+
+        pipe.set_progress_bar_config(disable=None)
+
+        output = pipe(**self.get_dummy_inputs(device))
+        image = output[0]
+        image = np.asarray(image)
+        image = image/255.0
+        image_slice = image[-3:, -3:, 0]
+
+        assert image.shape == (64, 64, 4)
+
+        expected_slice = np.array(
+            [0.5686274509803921, 0.5176470588235295, 0.6470588235294118, 0.44313725490196076, 0.6352941176470588, 0.4980392156862745, 0.396078431372549, 0.5529411764705883, 0.5411764705882353]
+        )
```

Comment:
Please keep these to four precisions only.

Reply:
Done